consoleme/lib/oidc.py

```diff
@@ -173,24 +179,51 @@ async def authenticate_user_by_oidc(request):
                 algorithm=algorithm,
             )
 
-            header = jwt.get_unverified_header(access_token)
-            key_id = header["kid"]
-            algorithm = header["alg"]
-            pub_key = oidc_config["jwt_keys"][key_id]
-            # This will raises errors if the audience isn't right or if the token is expired or has other errors.
-            decoded_access_token = jwt.decode(
-                access_token,
-                pub_key,
-                audience=config.get("get_user_by_oidc_settings.access_token_audience"),
-                algorithm=algorithm,
+            email = decoded_id_token.get(
+                config.get("get_user_by_oidc_settings.jwt_email_key", "email")
             )
+
+            # For google auth, the access_token does not contain JWT-parsable claims.
+            try:
+                header = jwt.get_unverified_header(access_token)
+                key_id = header["kid"]
+                algorithm = header["alg"]
```

Comment:
Have we ensured that we don't allow a none alg to get through?

Reply:
Done